server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/dto/DLPConfigurationItemDTOTest.java

```diff
@@ -0,0 +1,108 @@
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.dto;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.james.dlp.api.DLPConfigurationItem;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+
+class DLPConfigurationItemDTOTest {
+
+    private static final String ID = "id";
+    private static final String EXPRESSION = "expression";
+    private static final String EXPLANATION = "explanation";
+    private static final String NULL_ID = null;
+    private static final String NULL_EXPRESSION = null;
+
+    @Test
+    void toDTOsShouldBeSetAllFields() {
+        DLPConfigurationItemDTO dto = DLPConfigurationItemDTO.toDTO(
+            DLPConfigurationItem.builder()
+                .id(DLPConfigurationItem.Id.of(ID))
+                .expression(EXPRESSION)
+                .explanation(EXPLANATION)
+                .targetsSender(Optional.of(true))
+                .targetsRecipients(Optional.of(true))
+                .targetsContent(Optional.of(true))
+                .build());
+
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(dto.getId()).isEqualTo(ID);
+            softly.assertThat(dto.getExpression()).isEqualTo(EXPRESSION);
+            softly.assertThat(dto.getExplanation().get()).isEqualTo(EXPLANATION);
+            softly.assertThat(dto.getTargetsSender().get()).isTrue();
+            softly.assertThat(dto.getTargetsRecipients().get()).isTrue();
+            softly.assertThat(dto.getTargetsContent().get()).isTrue();
+        });
+    }
+
+    @Test
+    void toDLPConfigurationsShouldBeSetAllFields() {
+        List<DLPConfigurationItem> items = 
+            DLPConfigurationItemDTO.toDLPConfigurations(
+                ImmutableList.of(
+                    new DLPConfigurationItemDTO(ID,
+                            EXPRESSION,
+                            Optional.of(EXPLANATION),
+                            Optional.of(true),
+                            Optional.of(true),
+                            Optional.of(true))));
+
+        DLPConfigurationItem item = items.get(0);
```

Comment:
I really don't like this `.get(0)`. Can we avoid it, by allowing to transform single items, what you are really testing here?

Reply:
I removed `.get(0)`. 
See https://github.com/trantienduchn/james-project/blob/8eb85d292fb1aaaedc5cdc1feff394011b9cd549/server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/dto/DLPConfigurationItemDTOTest.java

Reply:
It's still on the full PR diff